File: repos_estudiantes_TARQUI_primera_entrega/ILIMORI_TOTOLA_ANEL_ESCARLET/PRACTICA 1/INTRODUCCION A POO/2/Bus.py
class Bus:
    def __init__(self, capacidad):
        self.pasajeros = 0
        self.capacidad = capacidad
        self.pasajes = 0.0

    def subir_pasajero(self, x):
        asientos_disponibles = self.capacidad - self.pasajeros
        if x > asientos_disponibles:
            print(f"No hay asientos disponibles. Solo hay {asientos_disponibles}")
            self.pasajeros += asientos_disponibles
        else:
            self.pasajeros += x
            print(f"Se subieron {x} pasajeros")
        print(f"Ahora hay {self.pasajeros} pasajeros en el bus")

    def cobrar_pasaje(self):
        costo_por_pasajero = 1.5
        cobro_total = self.pasajeros * costo_por_pasajero
        self.pasajes += cobro_total
        print(f"Se cobraron {cobro_total} Bs en pasajes")
        return cobro_total

    def asientos_disponibles(self):
        disponibles = self.capacidad - self.pasajeros
        print(f"Los asientos disponibles son {disponibles}")
        return disponibles

#MAIN
if __name__ == "__main__":
    puma = Bus(30)
    puma.asientos_disponibles()
    puma.subir_pasajero(10)
    puma.cobrar_pasaje()
    puma.asientos_disponibles()
    puma.subir_pasajero(50)
    puma.cobrar_pasaje()
    puma.asientos_disponibles()

File: repos_estudiantes_TARQUI_primera_entrega/PRADO_FLORES_ARACELY_NAOMI/POO ORIENTADA A OBJETOS/PooDos/Bus.java
package PooDos;

//Ejercicio2

public class Bus {
	private int pasajeros;
	private int capacidad;
	private double pasajes;

	public Bus (int capacidad) {
		this.pasajeros=0;
		this.capacidad = capacidad;
		this.pasajes= 0.0;
	}
		
	public void subirPasajero (int x) {
		int asientosDisponibles = capacidad - pasajeros;
		if (x > asientosDisponibles ) {
			System.out.println("No hay asientos disponibles. Solo hay "+ asientosDisponibles );
			pasajeros = pasajeros + asientosDisponibles;
		}else {
			pasajeros = pasajeros + x;
			System.out.println("se subieron " + x+ " pasjeros");
		}
		System.out.println("Ahora hay "+ pasajeros + " pasajeros en el bus");
	}
	public double cobrarPasaje () {
		double costoPorPasajero = 1.5;
		double cobroTotal = pasajeros * costoPorPasajero ;
		pasajes = pasajes + cobroTotal;
		System.out.println("Se cobraron "+cobroTotal+"Bs en pasajes" );
		return cobroTotal;
		
	}
	public int asientosDisponibles() {
		int Disponibles = capacidad - pasajeros;
		System.out.println("Los asientos disponibles son "+Disponibles);
		return Disponibles;
	}
	
	public static void main (String [] args) {
		Bus puma = new Bus(30);
		puma.asientosDisponibles();
		puma.subirPasajero(10);
		puma.cobrarPasaje();
		puma.asientosDisponibles();
		puma.subirPasajero(50);
		puma.cobrarPasaje();
		puma.asientosDisponibles();
	}
}